ShelterModule/Controllers/AnnouncementController.cs

```diff
@@ -0,0 +1,122 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using ShelterModule.Models.Announcements;
+using ShelterModule.Models.Pets;
+using ShelterModule.Services.Interfaces.Announcements;
+using ShelterModule.Services.Interfaces.Pets;
+using ShelterModule.Services.Interfaces.Shelters;
+
+namespace ShelterModule.Controllers;
+
+[ApiController]
+[Route("announcements")]
+public class AnnouncementController : ControllerBase
+{
+    private readonly IAnnouncementCommand _command;
+    private readonly IPetCommand _petCommand;
+    private readonly IPetQuery _petQuery;
+    private readonly IAnnouncementQuery _query;
+    private readonly IShelterQuery _shelterQuery;
+
+    public AnnouncementController(IAnnouncementQuery query, IAnnouncementCommand command, IPetQuery petQuery,
+        IPetCommand petCommand, IShelterQuery shelterQuery)
+    {
+        _query = query;
+        _command = command;
+        _petQuery = petQuery;
+        _petCommand = petCommand;
+        _shelterQuery = shelterQuery;
+    }
+
+    /// <summary>
+    ///     Returns an announcement with a given ID
+    /// </summary>
+    /// <param name="id"> ID of the announcement that should be returned </param>
+    /// <returns> Announcement with a given ID </returns>
+    [HttpGet]
+    [Route("{id:guid}")]
+    [ProducesResponseType(typeof(AnnouncementResponse), StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(ValidationProblemDetails), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(ValidationProblemDetails), StatusCodes.Status404NotFound)]
```

Comment:
We should have `typeof(NotFoundResponse)` here, since that's what is actually being returned.